packages/mdc-menu-surface/_mixins.scss

```diff
@@ -34,3 +76,72 @@
 @mixin mdc-menu-surface-shape-radius($radius, $rtl-reflexive: false) {
   @include mdc-shape-radius($radius, $rtl-reflexive);
 }
+
+
+//
+// Private
+//
+@mixin mdc-menu-surface-base_($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+  $feat-animation: mdc-feature-create-target($query, animation);
+
+  @include mdc-feature-targets($feat-structure) {
+    display: none;
+    position: absolute;
+    box-sizing: border-box;
+    max-width: calc(100vw - #{$mdc-menu-surface-min-distance-from-edge});
+    max-height: calc(100vh - #{$mdc-menu-surface-min-distance-from-edge});
+    margin: 0;
+    padding: 0;
+    transform: scale(1);
+    transform-origin: top left;
+    opacity: 0;
+    overflow: auto;
+    will-change: transform, opacity;
+    z-index: $mdc-menu-surface-z-index;
+
+    &:focus {
+      outline: none;
+    }
+
+    &--animating-open {
+      display: inline-block;
+      transform: scale(.8);
+      opacity: 0;
+    }
+
+    &--open {
+      display: inline-block;
+      transform: scale(1);
+      opacity: 1;
+    }
+
+    &--animating-closed {
+      display: inline-block;
+      opacity: 0;
+    }
+  }
+
+  @include mdc-feature-targets($feat-animation) {
+    &--animating-open {
```

Comment:
It may be worth covering `--animating-open` and `--animating-closed` separately so that we can nest the `feature-targets` invocations inside them. As written, this currently creates redundant selectors in the output that previously didn't exist.